include/kernel.h

```diff
@@ -493,6 +493,10 @@ struct _thread_base {
 	u8_t cpu_mask;
 #endif
 
+#ifdef CONFIG_NEWLIB_LIBC
```

Comment:
It would be desirable to add a separate newlib config symbol that enables `reent` support (e.g. `CONFIG_NEWLIB_LIBC_REENT`).

This symbol can be `default y if MULTITHREADING` so as to only enable `reent` support when multi-threading is enabled.

Reply:
If there is definitelly only one thread running I think that's a good idea.

Reply:
yeah this would be good, we have some use-cases for disabling mulithreading and using Zephyr more like a HAL (bootloaders, for example)

Reply:
How do you think about about the other switch to reserve memory for the locks? Is it ok, to have one? Would you set the default value to 0? Actually, I dont like to set it to 0 as only the malloc_lock hooks work in this szenario, but I was concerned that existing applications could run out of memory in case I choose a too little default value